File: 98-validate-binary-search-tree/98-validate-binary-search-tree.py
class Solution:
    def recurse(self, root : Optional[TreeNode], arr : List[int]):
        if root.left != None: self.recurse(root.left, arr)
        arr.append(root.val)
        if root.right != None: self.recurse(root.right, arr)
    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        arr = []
        self.recurse(root, arr)
        for i in range(1, len(arr)):
            if arr[i-1] >= arr[i]:
                return False
        return True

File: 98-validate-binary-search-tree/98-validate-binary-search-tree.java
class Solution {
    boolean flag;
    void dfs(TreeNode node, Stack<Integer> list) {
        if(node.left != null) dfs(node.left, list);
        if(!list.isEmpty()) {
            if(list.peek() >= node.val) flag = false;
        }
        list.push(node.val);
        if(node.right != null) dfs(node.right, list);
    }
    public boolean isValidBST(TreeNode root) {
        flag = true;
        Stack<Integer> list = new Stack();
        dfs(root, list);
        return flag;
    }
}